firrtl/src/main/scala/firrtl/ir/IR.scala

```diff
@@ -194,6 +194,13 @@ abstract class Literal extends Expression {
   val value: BigInt
   val width: Width
 }
+
+case class ProbeExpr(expr: Expression) extends Expression with UseSerializer {
+  def tpe = ProbeType(expr.tpe)
```

Comment:
As proposed, `probe(e)` is passive version of the input (which may have mixed alignment).
Turns out today we don't support taps of non-passive things anyway (https://github.com/llvm/circt/pull/4815, FWIW, building on new ref bits in CIRCT), but this seems important to model.

Reply:
Hmmm given what @jackkoenig said in https://github.com/chipsalliance/chisel/pull/3088#discussion_r1139162397, I think this maybe should also get checked from whatever Chisel API actually calls `ProbeExpr()`.

Reply:
This isn't a check/requirement it's what the type of a probe-of-a-non-passive-type is, aka it's how the type of the expression is computed (vs the linked comment is about constraining where the expression can be used).

In the short term just rejecting probes of non-passive (even though supported in the spec) would work, but that's not because it's a requirement but because we wouldn't have implemented computing the type of such a probe expression in Chisel(/here?) yet.  Probably not a priority since it's not something we support at all today re:tapping mixed-alignment (at least from what CIRCT sees).

I guess I'm asking if this could be: `def tpe = ProbeType(expr.tpe.toPassive())` (if such existed)

Anyway this seems hard and don't mean to be difficult.  As pieces are landed, especially in CIRCT, we/I can revisit this :+1: .  I only said anything because if we're saying what the type of a probe of an expression is, it's something like the above and not just boxing it with `Probe`/`RWProbe`.  I'm not sure how/if this type will ever be inspected/produced/used as-is anyway... unless there's type-checking done (e.g., ensuring compatible connect's) which sounds like maybe is not what this layer is for.

Reply:
I have implemented passive-type coercion using `Output()` as @azidar suggested!